modeshape-jcr/src/main/java/org/modeshape/jcr/security/acl/AccessControlEntryImpl.java

```diff
@@ -158,6 +156,16 @@ public boolean equals( Object other ) {
 
     @Override
     public int hashCode() {
-        return this.principal.hashCode();
+        int result = principal.hashCode();
+        result = 31 * result + privileges.hashCode();
+        return result;
```

Comment:
Or maybe just the following?

    return HashCode.compute(principal,privileges);